### PR TITLE
[INTEL MKL] Optimization for MirrorPad op.

### PR DESCRIPTION
**Pull request by Solaryee**
In the original code the index was not being checked correctly for SIMD vectorization. For example, if it is a row-major matrix, the index is correctly checked for vectorization (packets) only for the first row. We need to use `%` operator to make sure the index is correct for each row.

Also, several test cases are added to test the correctness and performance. Some outputs of benchmark test:

| test case|previous|optimized|
| ------ | ------ |------ |
|BM_MirrorPad_cpu_1_16_16_32_1| 1.9M items/s  |6.2M items/s|
|BM_MirrorPad_cpu_1_16_16_32_8| 10.0M items/s  |13.4M items/s|
|BM_MirrorPad_cpu_1_512_512_16_1| 106.1M items/s  |422.1M items/s|
|BM_MirrorPad_cpu_1_512_512_16_256| 118.7M items/s | 196.9M items/s|

### CODE REVIEW

**Comment by ShengYang1** [comment]
Any updates?

**Comment by ShengYang1** [comment]
Any update? @penpornk

**Review by penpornk — APPROVED** [review]
I'm so sorry for the delay! Nice catch and thank you very much for the PR!

File: tensorflow/core/kernels/mirror_pad_op_test.cc [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  Tensor expected(allocator(), DT_FLOAT, TensorShape({1, kOutput, kOutput, 1}));
+  test::FillFn<float>(&expected, [](int i) -> float {
+    i = i % kOutput;
+    if (0 <= i && i < kPad)
+      return kPad - i;
+    else if (kPad <= i && i < kInput + kPad)
+      return i - kPad;
+    else if (kInput + kPad <= i && i < kOutput)
+      return 2 * kInput + kPad - 2 - i;
+  });
```
Comment (penpornk) [inline tensorflow/core/kernels/mirror_pad_op_test.cc]: `error: non-void lambda does not return a value in all control paths [-Werror,-Wreturn-type]`

File: tensorflow/core/kernels/mirror_pad_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  Tensor expected(allocator(), DT_FLOAT, TensorShape({1, kOutput, kOutput, 1}));
+  test::FillFn<float>(&expected, [](int i) -> float {
+    i = i % kOutput;
+    if (0 <= i && i < kPad)
+      return kPad - i - 1;
+    else if (kPad <= i && i < kInput + kPad)
+      return i - kPad;
+    else if (kInput + kPad <= i && i < kOutput)
+      return 2 * kInput + kPad - 1 - i;
+  });
```
Comment (penpornk) [inline tensorflow/core/kernels/mirror_pad_op_test.cc]: `error: non-void lambda does not return a value in all control paths [-Werror,-Wreturn-type]`

**Review by penpornk — CHANGES_REQUESTED** [review]
Could you please help fix clang build errors? Thank you!

**Comment by ShengYang1** [comment]
@penpornk  Fixed.

**Review by penpornk — APPROVED** [review]
Thank you and happy holidays!

**Pull request merged**